src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/TransmissionStorage.cs

```diff
@@ -139,24 +140,34 @@ public virtual Transmission Dequeue()
                         return transmission;
                     }
                 }
-                catch (UnauthorizedAccessException)
+                catch (UnauthorizedAccessException uae)
                 {
                     if (file == null)
                     {
-                        return null; // because the process does not have permission to access the folder
+                        TelemetryChannelEventSource.Log.TransmissionStorageDequeueUnauthorizedAccessException(this.folder.Name, uae.ToString());
```

Comment:
In this case there is no file, it is null.
I'm not convinced that this can actually happen but the original logic implied that it can so in keeping with the original logic I'm logging what does exist in accordance with the original comment.